pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -31,10 +41,82 @@ import (
 	"istio.io/istio/pkg/log"
 )
 
-// Endpoints implements MeshDiscovery.Endpoints()
-func Endpoints(ds *v1.DiscoveryService, serviceClusters []string) *xdsapi.DiscoveryResponse {
+// EDS returns the list of endpoints (IP:port and in future labels) associated with a real
+// or virtual service. A virtual service is a subset of a real service, selected using labels.
```

Comment:
nit: please rephrase to "associated with a service or its subset". We already have a networking CRD called virtual service that has a different meaning, and people reading this code would get confused.

Reply:
Updated.